packages/admin/src/pages/super-admin-dashboard/user/[id]/schemes/[schemeId]/change-owner.page.tsx

```diff
@@ -0,0 +1,120 @@
+import { Button, FlexibleQuestionPageLayout, TextInput } from 'gap-web-ui';
+import CustomLink from '../../../../../../components/custom-link/CustomLink';
+import Meta from '../../../../../../components/layout/Meta';
+import InferProps from '../../../../../../types/InferProps';
+import { GetServerSidePropsContext } from 'next';
+import {
+  getSessionIdFromCookies,
+  getUserTokenFromCookies,
+} from '../../../../../../utils/session';
+import { checkNewAdminEmailIsValid } from '../../../../../../services/UserService';
+import QuestionPageGetServerSideProps from '../../../../../../utils/QuestionPageGetServerSideProps';
+
+type PageBodyResponse = {
+  emailAddress: string;
+};
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const { id: userId, schemeId } = context.params as Record<string, string>;
+  const oldEmailAddress = decodeURIComponent(
+    context.query.oldEmailAddress as string
+  );
+  const newEmailAddress = context.query.newEmailAddress
+    ? decodeURIComponent(context.query.newEmailAddress as string)
+    : null;
+  const schemeName = decodeURIComponent(context.query.schemeName as string);
+
+  async function handleRequest(body: PageBodyResponse, jwt: string) {
+    await checkNewAdminEmailIsValid(
+      jwt,
+      getUserTokenFromCookies(context.req),
+      body.emailAddress
+    );
+    return body.emailAddress;
+  }
+
+  async function fetchPageData() {
+    return {
+      userId: userId ?? null,
+      schemeName,
+      prevEmailAddress: newEmailAddress,
+    };
+  }
+
+  return QuestionPageGetServerSideProps<
+    PageBodyResponse,
+    Awaited<ReturnType<typeof fetchPageData>>,
+    Awaited<ReturnType<typeof handleRequest>>
+  >({
+    context,
+    fetchPageData,
+    handleRequest,
+    jwt: getSessionIdFromCookies(context.req),
+    onErrorMessage: 'Failed to change scheme owner.',
+    onSuccessRedirectHref: (emailAddress) =>
+      `/super-admin-dashboard/user/${userId}/schemes/${schemeId}/confirm-change-owner?newEmailAddress=${encodeURIComponent(
+        emailAddress
+      )}&oldEmailAddress=${encodeURIComponent(
+        oldEmailAddress
+      )}&schemeName=${encodeURIComponent(schemeName)}`,
```

Comment:
yeah can we write a helper method to build this url? its uggo

Reply:
yee good shout